Allow passing extra arguments along in lean-language-server

Julian opened the pull request:
Does so by allowing writing e.g.:

    $ lean-language-server --stdio -- -M 4096 -T 10000

which will provide the extra `argv` down to Lean.

This is 'wrong' in the sense that it should really use the LSP-native
way to do so, which is responding to `workspace/configuration` requests,
but the current version of lean-language-server uses a quite old version
of vscode-languageserver, and I couldn't get it to work on a newer one
without significant changes. For another time...

Julian commented:
Any thoughts on this PR? It's not ideal, but without it, the default memory limit I think is low enough that everyone will hit it pretty quickly in a mathlib file. If this looks reasonable though we could up that default to match VSCode's.

bryangingechen commented:
Ah, I wasn't aware `lean-language-server` was actually being used, but it looks like it's used for [`lean.nvim`](https://github.com/Julian/lean.nvim)!

gebner commented:
Thanks for pinging!

Julian commented:
Hooray, thanks both!

> Ah, I wasn't aware lean-language-server was actually being used, but it looks like it's used for lean.nvim!

Yep!

Julian commented:
It seems I must have broken something, though not sure quite how.

But installing `lean-language-server` now gives something that doesn't have a shebang in it? So it doesn't get run with `node`.

```
~/Development/lean.nvim is a git repository on main 
⊙  npm install -g lean-language-server                                                                                                                                              julian@Airm

added 9 packages, and audited 11 packages in 654ms

found 0 vulnerabilities

~/Development/lean.nvim is a git repository on main 
⊙  head =lean-language-server                                                                                                                                                       julian@Airm
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const lean_client_js_node_1 = require("lean-client-js-node");
const vscode_languageserver_1 = require("vscode-languageserver");
const vscode_uri_1 = require("vscode-uri");
const connection = vscode_languageserver_1.createConnection();
const server = new lean_client_js_node_1.Server(null);
server.error.on((err) => {
    switch (err.error) {
        case 'stderr':

~/Development/lean.nvim is a git repository on main 
⊙  lean-language-server --stdio                                                                                                                                                     julian@Airm
/Users/julian/.local/bin/lean-language-server: line 1: use strict: command not found
/Users/julian/.local/bin/lean-language-server: line 2: syntax error near unexpected token `exports,'
/Users/julian/.local/bin/lean-language-server: line 2: `Object.defineProperty(exports, "__esModule", { value: true });'

~/Development/lean.nvim is a git repository on main 
⊙  node =lean-language-server --stdio                                                                                                                                               julian@Airm
<works>
```

gebner commented:
Did it have that before?

Julian commented:
Yes, seems so, so I assume maybe it's a change in the typescript compiler's behavior? I had to jump back on some calls, but will keep having a look, but rolling back "fixes" it:

```
~/Development/lean-client-js/lean-language-server is a git repository on extra-args 
⊙  npm install -g lean-language-server@2.0.1                                                                                                                                      julian@Airm ●
^R     ░░░░░░░░░░░░⸩ ⠏ idealTree:vscode-languageserver-protocol: sill placeDep node_modules/lean-language-server vscode-languageserver-protocol@3.5.1 OK for: vscode-languageserver@3.5.1 want:

added 9 packages, and audited 11 packages in 4s

found 0 vulnerabilities

~/Development/lean-client-js/lean-language-server is a git repository on extra-args 
⊙  lean-language-server --stdio                                                                                                                                                   julian@Airm ●
^C

~/Development/lean-client-js/lean-language-server is a git repository on extra-args 
⊙  head =lean-language-server                                                                                                                                                     julian@Airm ●
#!/usr/bin/env node
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const lean_client_js_node_1 = require("lean-client-js-node");
const vscode_languageserver_1 = require("vscode-languageserver");
const vscode_uri_1 = require("vscode-uri");
const connection = vscode_languageserver_1.createConnection();
const server = new lean_client_js_node_1.Server(null);
server.error.on((err) => {
    switch (err.error) {

~/Development/lean-client-js/lean-language-server is a git repository on extra-args 
⊙  npm install -g lean-language-server@2.0.3                                                                                                                                      julian@Airm ●

changed 9 packages, and audited 11 packages in 671ms

found 0 vulnerabilities

~/Development/lean-client-js/lean-language-server is a git repository on extra-args 
⊙  head =lean-language-server                                                                                                                                                     julian@Airm ●
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const lean_client_js_node_1 = require("lean-client-js-node");
const vscode_languageserver_1 = require("vscode-languageserver");
const vscode_uri_1 = require("vscode-uri");
const connection = vscode_languageserver_1.createConnection();
const server = new lean_client_js_node_1.Server(null);
server.error.on((err) => {
    switch (err.error) {
        case 'stderr':

```
